hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/AbstractRemoveModelNodeCommand.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.agent.monitor.cmd;
+
+import org.hawkular.agent.monitor.extension.MonitorServiceConfiguration;
+import org.hawkular.agent.monitor.inventory.InventoryIdUtil;
+import org.hawkular.agent.monitor.inventory.InventoryIdUtil.ResourceIdParts;
+import org.hawkular.agent.monitor.inventory.ManagedServer;
+import org.hawkular.agent.monitor.inventory.Name;
+import org.hawkular.agent.monitor.inventory.dmr.DMRInventoryManager;
+import org.hawkular.agent.monitor.inventory.dmr.LocalDMRManagedServer;
+import org.hawkular.agent.monitor.log.AgentLoggers;
+import org.hawkular.agent.monitor.log.MsgLogger;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.bus.common.BinaryData;
+import org.hawkular.cmdgw.api.MessageUtils;
+import org.hawkular.cmdgw.api.ResourcePathRequest;
+import org.hawkular.cmdgw.api.ResourcePathResponse;
+import org.hawkular.cmdgw.api.ResponseStatus;
+import org.hawkular.dmrclient.Address;
+import org.hawkular.dmrclient.DatasourceJBossASClient;
+import org.hawkular.inventory.api.model.CanonicalPath;
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public abstract class AbstractRemoveModelNodeCommand<REQ extends ResourcePathRequest, RESP extends ResourcePathResponse>
+        implements Command<REQ, RESP> {
+    private static final MsgLogger log = AgentLoggers.getLogger(AbstractRemoveModelNodeCommand.class);
+
+    private final String entityType;
+
+    public AbstractRemoveModelNodeCommand(String entityType) {
+        super();
+        this.entityType = entityType;
+    }
+
+    @Override
+    public BasicMessageWithExtraData<RESP> execute(REQ request, BinaryData ignored, CommandContext context)
+            throws Exception {
+
+        String rawResourcePath = request.getResourcePath();
+        log.infof("Received request to remove [%s] given by inventory path [%s]", entityType, rawResourcePath);
+
+        MonitorServiceConfiguration config = context.getMonitorServiceConfiguration();
+
+        // Based on the resource ID we need to know which inventory manager is handling it.
+        // From the inventory manager, we can get the actual resource.
+        CanonicalPath canonicalPath = CanonicalPath.fromString(rawResourcePath);
+
+        String resourceId = canonicalPath.ids().getResourcePath().getSegment().getElementId();
+        ResourceIdParts idParts = InventoryIdUtil.parseResourceId(resourceId);
+        String modelNodePath = idParts.getIdPart();
+
+        RESP response = createResponse();
+        MessageUtils.prepareResourcePathResponse(request, response);
+
+        DatasourceJBossASClient dsc = null;
```

Comment:
I see this  AbstractRemoveModelNodeCommand class is abstract and thus I assume meant for reuse - however, it is using the very datasource-specific class of DatasourceJBossASClient. Why is that?